src/transformers/modeling_tf_bert.py

```diff
@@ -1028,16 +1341,31 @@ def call(
     """Bert Model with a `next sentence prediction (classification)` head on top. """,
     BERT_START_DOCSTRING,
 )
-class TFBertForNextSentencePrediction(TFBertPreTrainedModel):
+class TFBertForNextSentencePrediction(TFBertPreTrainedModel, TFNextSentencePredictionLoss):
+    authorized_unexpected_keys = [r"predictions"]
+
     def __init__(self, config, *inputs, **kwargs):
         super().__init__(config, *inputs, **kwargs)
 
-        self.bert = TFBertMainLayer(config, name="bert")
-        self.nsp = TFBertNSPHead(config, name="nsp___cls")
+        self.bert = TFBertMainLayer(config=config, name="bert")
+        self.cls = TFBertOnlyNSPHead(config=config, name="cls")
```

Comment:
Weight parameter naming should be avoided!!! Changing `self.nsp` to `self.cls` means that all models trained < 4.0 cannot be used with transformers 4.0. The renaming is definitely not worth such a HUGE breaking change

Reply:
Okey @jplu just showed me the regex in the `load_tf_weights` function that make this backwards compatible. Still not sure if that's the way to go given that adding regexes for all models could explode the complexity and would make such a super important function like `load_tf_weights` very much unreadable. Think I'd rather vote against this IMO more "cosmetic" name change. (cc. @LysandreJik @sgugger )

Reply:
Let's see how many exceptions we can find between PT and TF models in the naming, if there are not much it should be largely manageable, as sharing the same name helps quite a lot I think the PT users to understand the TF models and vice versa. Otherwise, if there are way too many changes we can revert this.